shell/platform/embedder/embedder_render_target_cache.cc

```diff
@@ -10,63 +10,41 @@ EmbedderRenderTargetCache::EmbedderRenderTargetCache() = default;
 
 EmbedderRenderTargetCache::~EmbedderRenderTargetCache() = default;
 
-std::pair<EmbedderRenderTargetCache::RenderTargets,
-          EmbedderExternalView::ViewIdentifierSet>
-EmbedderRenderTargetCache::GetExistingTargetsInCache(
-    const EmbedderExternalView::PendingViews& pending_views) {
-  RenderTargets resolved_render_targets;
-  EmbedderExternalView::ViewIdentifierSet unmatched_identifiers;
-
-  for (const auto& view : pending_views) {
-    const auto& external_view = view.second;
-    if (!external_view->HasEngineRenderedContents()) {
-      continue;
-    }
-    auto& compatible_targets =
-        cached_render_targets_[external_view->CreateRenderTargetDescriptor()];
-    if (compatible_targets.empty()) {
-      unmatched_identifiers.insert(view.first);
-    } else {
-      std::unique_ptr<EmbedderRenderTarget> target =
-          std::move(compatible_targets.top());
-      compatible_targets.pop();
-      resolved_render_targets[view.first] = std::move(target);
-    }
+std::unique_ptr<EmbedderRenderTarget>
+EmbedderRenderTargetCache::GetRenderTarget(
+    const EmbedderExternalView::RenderTargetDescriptor& descriptor) {
+  auto compatible_target = cached_render_targets_.find(descriptor);
+  if (compatible_target == cached_render_targets_.end()) {
+    return nullptr;
+  } else {
```

Comment:
pedantic style nit: just leave the `if` as a bail-out conditional, then make the contents of the `else` the main function body. Avoids a teensy bit of indentation, but also slightly defensive against any future change that would leave the top-level without a `return` statement.